frontend/src/components/common/form/ParentSelect.tsx

```diff
@@ -136,28 +137,30 @@ export const ParentSelect: React.FC<IParentSelect> = ({
             }),
             x => x.parentId,
           );
-          const items = Object.keys(parents)
+          const items = Object.keys(resultGroup)
             .sort()
-            .map(parent => (
-              <Fragment key={parent}>
-                {!!results.find((x: SelectOption) => x.value === parent) && (
+            .map(parentId => (
+              <Fragment key={parentId}>
+                {!!results.find((x: SelectOption) => x.parentId === +parentId) && (
                   <Menu.Header
                     onClick={() =>
                       enableMultiple
                         ? handleMultiSelectHeaderClick(
-                            results.filter(x => x.parentId?.toString() === parent),
+                            results.filter(x => x.parentId?.toString() === parentId),
```

Comment:
its a bit confusing how here we are doing string comparison of parentIds but above we are doing number comparisons.

Reply:
Oh woops changed one and forgot to do the other